src/Sentry/Protocol/Metrics/Metric.cs

```diff
@@ -73,7 +114,7 @@ public async Task SerializeAsync(Stream stream, IDiagnosticLogger? logger, Cance
 
         foreach (var value in SerializedStatsdValues())
         {
-            await Write($":{value.ToString(CultureInfo.InvariantCulture)}");
+            await Write($":{((IConvertible)value).ToString(CultureInfo.InvariantCulture)}");
```

Comment:
If user's implemented a custom Metric instrument and tried to serlize this, using values that don't implement `IConvertible` then they could break it. I can't see users doing that though, since Sentry.io doesn't support custom Metric instruments... there's be no reason to create such a class. 

Ideally we'd use `IConvertible` in the interface but it's not CLS Compliant.

Reply:
It feels like a pitfall. Auers will for sure return random types here and break it. Can we define a type ourselves to represent this?